### ISSUE TITLE
Add posibility tu use different activation function on output neurons and add possibility of batch training

### ISSUE THREAD

**Issue by willayy** [willayy/SNNPy#18]

**Comment by willayy** [comment]
- [ ] Divide activations into hidden activation and output activation, add Softmax and Linear activation.
- [ ] Make forward propogation return the output, make fit use a passed argument as predicted value instead of looking in the output neurons to make batch training possible.
- [ ] Test backprop and forward prop
- [ ] Finally get something to converge properly lol.

**Comment by willayy** [comment]
This is fixed, even added specific activations for input neurons
